kedro-datasets/kedro_datasets_experimental/langchain/_cohere.py

```diff
@@ -0,0 +1,73 @@
+"""
+Cohere dataset definition.
+"""
+
+from typing import Any, NoReturn
+
+from kedro.io import AbstractDataset, DatasetError
+from langchain_cohere import ChatCohere
```

Comment:
I think it's even better to switch to `ChatCohere` instead of `Cohere` as the last seems to be deprecated: https://api.python.langchain.com/en/latest/llms/langchain_community.llms.cohere.Cohere.html

@merelcht so the error you've experienced could be because of versions mismatch